### DISCUSSION TITLE
Background task capture exception

### DISCUSSION THREAD

**Discussion by fishsixs** [fastapi/fastapi#12088] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
from fastapi import FastAPI, BackgroundTasks
from fastapi.responses import JSONResponse
from starlette.requests import Request

app = FastAPI()


class CustomException(Exception):

    def __init__(
            self,
            message: str,

    ):
        self.message = message


@app.exception_handler(CustomException)
async def custom_exception_handler(request: Request, exc: CustomException):
    """
    自定义异常
    """
    return JSONResponse(
        status_code=200,
        content={
            "message": exc.message
        },
    )


def task_func():
    raise CustomException("dd")


@app.post("/")
async def report_batch_ahu(
        background_tasks: BackgroundTasks
):
    background_tasks.add_task(task_func)
    return {"status": "success"}


if __name__ == '__main__':
    import uvicorn

    uvicorn.run(app, host="0.0.0.0", port=8000)
```

### Description

How do I respond to exceptions when testing background tasks?
Result error：RuntimeError: Caught handled exception, but response already started.
```python
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\venv\ekdcc_cad_py311\Lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 404, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\venv\ekdcc_cad_py311\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\venv\ekdcc_cad_py311\Lib\site-packages\fastapi\applications.py", line 1115, in __call__
    await super().__call__(scope, receive, send)
  File "D:\venv\ekdcc_cad_py311\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "D:\venv\ekdcc_cad_py311\Lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "D:\venv\ekdcc_cad_py311\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "D:\venv\ekdcc_cad_py311\Lib\site-packages\starlette\middleware\exceptions.py", line 83, in __call__
    raise RuntimeError(msg) from exc
RuntimeError: Caught handled exception, but response already started.
```
How can I deal with it

### Operating System

Windows

### FastAPI Version

0.104.0

### Pydantic Version

2.5.3

### Python Version

Python 3.11.6

**Comment by YuriiMotov** [comment, marked as answer]
`@app.exception_handler` is for handling exceptions in the endpoint functions and in router.

You can not send any response as a reaction to an error during the execution of background task, because FastAPI executes background tasks after sending the HTTP response. You can't modify response that has already been sent.

**Reply by rvishruth** [reply]
@YuriiMotov answer solves this question. But this might help: https://fastapi.tiangolo.com/tutorial/background-tasks/
